wcomponents-test-lib/src/main/java/com/github/bordertech/wcomponents/test/selenium/by/ByButtonText.java

```diff
@@ -0,0 +1,85 @@
+package com.github.bordertech.wcomponents.test.selenium.by;
+
+import java.util.List;
+import org.openqa.selenium.By;
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.internal.FindsByXPath;
+
+/**
+ * Finds a button by the text value on the button.
+ */
+public class ByButtonText extends By {
+
+
+	/**
+	 * Button's text value by exact root.
+	 */
+	public static final String XPATH_BUTTON_VALUE_EXACT_ROOT = "//button[text()='%1$s']";
```

Comment:
These XPath selectors will all fail if the button has element children which contain the text. For example a WMenuItem or a WButton with an image and imageposition. To allow for this you should use `"//button[.//text()='%1$s']"` and similar for the other selectors.